pkg/testutil/nerdtest/requirements.go

```diff
@@ -19,125 +19,293 @@ package nerdtest
+			// We do not even want to get to the setup phase here
+			helpers.Write(flaky, only)
+			ret = environmentIsForFlaky()
+			if !ret {
+				mess = "runner skips flaky compatible tests in the non-flaky environment"
+			}
+			return ret, mess
+		},
 	}
-	return ret, mess
-})
-
-var NerdctlNeedsFixing = test.MakeRequirement(func(data test.Data, t *testing.T) (ret bool, mess string) {
-	ret = testutil.GetTarget() == testutil.Docker
-	if ret {
-		mess = "current target is docker"
-	} else {
-		mess = "current target is nerdctl, but it is currently broken and not working for this"
+}
+
+// Docker marks a test as suitable solely for Docker and not Nerdctl
+// Generally used as test.Not(nerdtest.Docker), which of course it the opposite
+var Docker = &test.Requirement{
+	Check: func(data test.Data, helpers test.Helpers) (ret bool, mess string) {
+		ret = getTarget() == targetDocker
+		if ret {
+			mess = "current target is docker"
+		} else {
+			mess = "current target is not docker"
+		}
+		return ret, mess
+	},
+}
+
+// NerdctlNeedsFixing marks a test as unsuitable to be run for Nerdctl, because of a specific known issue which
+// url must be passed as an argument
+var NerdctlNeedsFixing = func(issueLink string) *test.Requirement {
+	return &test.Requirement{
+		Check: func(data test.Data, helpers test.Helpers) (ret bool, mess string) {
+			ret = getTarget() == targetDocker
+			if ret {
+				mess = "current target is docker"
+			} else {
+				mess = "current target is nerdctl, but we will skip as nerdctl currently has issue: " + issueLink
+			}
+			return ret, mess
+		},
 	}
-	return ret, mess
-})
-
-var Rootless = test.MakeRequirement(func(data test.Data, t *testing.T) (ret bool, mess string) {
-	// Make sure we DO not return "IsRootless true" for docker
-	ret = testutil.GetTarget() != testutil.Docker && rootlessutil.IsRootless()
-	if ret {
-		mess = "environment is rootless"
-	} else {
-		mess = "environment is rootful"
+}
+
+// BrokenTest marks a test as currently broken, with explanation provided in message, along with
+// additional requirements / restrictions describing what it can run on.
+var BrokenTest = func(message string, req *test.Requirement) *test.Requirement {
+	return &test.Requirement{
+		Check: func(data test.Data, helpers test.Helpers) (bool, string) {
+			ret, mess := req.Check(data, helpers)
+			return ret, message + "\n" + mess
+		},
+		Setup:   req.Setup,
+		Cleanup: req.Cleanup,
 	}
-	return ret, mess
-})
-
-var Build = test.MakeRequirement(func(data test.Data, t *testing.T) (ret bool, mess string) {
-	// FIXME: shouldn't we run buildkitd in a container? At least for testing, that would be so much easier than
-	// against the host install
-	ret = true
-	mess = "buildkitd is enabled"
-	if testutil.GetTarget() == testutil.Nerdctl {
-		_, err := buildkitutil.GetBuildkitHost(testutil.Namespace)
-		if err != nil {
+}
+
+// RootLess marks a test as suitable only for the rootless environment
+var RootLess = &test.Requirement{
```

Comment:
mayeb s/RootLess/Rootless/

Reply:
Yeah, I was wondering about that...

I am happy either way - I will take another look at that aspect (RootLess, RootFul) with the next round then.